### PR TITLE
Examples reorg

### PR DESCRIPTION
**Pull request by bryevdv**
This PR partially implements the plan laid out in 

https://github.com/bokeh/bokeh/wiki/IA-for-Bokeh-demos-and-examples

In particular:

* `compat` directory is collapsed
 - add "interesting" MPL examples
* `glyphs` -> `models`
* `interactions` -> `howto`

This PR also moves "testlike" mpl and `glyphs.ipynb` to `bokeh/tests` where we can soon add then to strict image diff testing. 

Right now there are netbooks in: `charts/notebook`, `plotting/notebook` and `interactive`. I'd like to handle the notebook situations later, coordinating migrations to `bokeh-notebooks` plus appropriate links to them from docs, in a separate PR.

### CODE REVIEW

**Comment by bryevdv** [comment]
ping @bokeh/core

**Comment by bryevdv** [comment]
@bokeh/core would like to merge this tomorrow.

File: examples/README.md [hunk 1]
```diff
@@ -12,9 +12,9 @@ This directory contains examples of Bokeh Apps, which are simple and easy to cre
 
 ### [`charts`](https://github.com/bokeh/bokeh/tree/master/examples/compat)
```
Comment (birdsarah) [inline examples/README.md]: I'd like to propose changing these to relative links. (e.g. https://github.com/bokeh/bokeh/commit/0d0b62b7aebbfa0f94180ac6de9e3d023e0eca94). That way, if examples change over time, people will be pointing to the correct branch.  I think this is particularly true, it we wanted to link to a release tagged copy of this README

Comment (birdsarah) [inline examples/README.md]: I just noticed in the full version of this readme (shakes fist at diffs) this note 

> "NOTE: All of the links below are to the "master" branch of the repository, which always contains the newest and most updated code. For questions about specific versions of Bokeh, please take care to refer to the correct branch for your version!"  

I would also propose removing that note I guess. Or reframing it as make sure the version of the examples matches the version of bokeh you're running.

Comment (bryevdv) [inline examples/README.md]: Sure, can you make the changes and push them?

File: examples/README.md [hunk 2]
```diff
@@ -25,13 +25,13 @@ This directory contains examples of plots that use Bokeh's [compatibility layer]
 
 This directory includes examples that show how to embed Bokeh plots in as DOM objects within an HTML document.
```
Comment (birdsarah) [inline examples/README.md]: > This directory includes examples that show how to embed Bokeh plots in as DOM objects within an HTML document.

propose change to

This directory includes examples that show how to embed Bokeh plots in an HTML document.

Comment (bryevdv) [inline examples/README.md]: sure

Comment (bryevdv) [inline examples/README.md]: maybe "plots and widgets"

File: examples/README.md [hunk 3]
```diff
@@ -25,13 +25,13 @@ This directory contains examples of plots that use Bokeh's [compatibility layer]
...
 
-This directory contains examples that use the lowest-level [`bokeh.models`](http://bokeh.pydata.org/en/latest/docs/user_guide/concepts.html#bokeh-models) interface. This interface mirrors the models found in the BokehJS browser library.
+The examples in this directory are mini-tutorials that demonstrate and explain some  particular aspect of Bokeh capability (such as [linking and brushing](http://www.infovis-wiki.net/index.php?title=Linking_and_Brushing)), or walk through a particular example in additional detail.
 
-### [`interactions`](https://github.com/bokeh/bokeh/tree/master/examples/interactions)
+### [`models`](https://github.com/bokeh/bokeh/tree/master/examples/models)
 
-The examples in this directory show off particular aspects of Bokeh user interaction, such as [linking and brushing](http://www.infovis-wiki.net/index.php?title=Linking_and_Brushing) or interactive hover annotations.
+This directory contains examples that use the lowest-level [`bokeh.models`](http://bokeh.pydata.org/en/latest/docs/user_guide/concepts.html#bokeh-models) interface. This interface mirrors the models found in the BokehJS browser library.
```
Comment (birdsarah) [inline examples/README.md]: I would scrap "This interface mirrors the models found in the BokehJS browser library." And make more of the link. e.g. 

This directory contains examples that use the lowest-level [`bokeh.models`](http://bokeh.pydata.org/en/latest/docs/reference/models.html) interface. For more understanding of the bokeh.models interface see [the concepts section of the user_guide](http://bokeh.pydata.org/en/latest/docs/user_guide/concepts.html#bokeh-models)

File: examples/README.md [hunk 4]
```diff
@@ -41,7 +41,7 @@ This directory contains example using the [`bokeh.plotting`](http://bokeh.pydata
 
 * `notebook` examples that display inline in [Jupyter](http://jupyter.org) notebooks
 
-* `server` examples that send output to a Bokeh server, using the python client library for the server 
+* `server` examples that send output to a Bokeh server, using the python client library for the server
```
Comment (birdsarah) [inline examples/README.md]: charts also has this same seperation - should we have this list higher up the readme.

Comment (bryevdv) [inline examples/README.md]: It won't, soon.

File: examples/README.md (hunk 3 above)
Comment (bryevdv) [inline examples/README.md]: Sounds good (please feel free to just make all these edits)

**Comment by birdsarah** [comment]
Pushed one small tweak to the gapminder example which was a dead link to the bokeh logo.

I could make the changes to the README if you wanted me to otherwise LGTM.

File: examples/README.md (hunk 4 above)
Comment (birdsarah) [inline examples/README.md]: how soon? (in this PR?)

Comment (bryevdv) [inline examples/README.md]: no but in the next few days

**Comment by bryevdv** [comment]
Yes please make the changes and then we can merge tomorrow

**Pull request merged**